deku-derive/src/macros/deku_read.rs

```diff
@@ -453,7 +453,7 @@ fn emit_bit_byte_offsets(
         || byte_offset.is_some()
     {
         Some(quote! {
-            let __deku_bit_offset = usize::try_from(__deku_input_bits.offset_from(__deku_rest))?;
+            let __deku_bit_offset = usize::try_from(unsafe { __deku_rest.as_bitptr().offset_from(__deku_input_bits.as_bitptr()) } )?;
```

Comment:
Lets add a `// SAFETY` for proving that this is fine, something like: https://github.com/bitvecto-rs/bitvec/blob/a7a82c67206d6e3ced67294883f6d1baaae7d74c/src/ptr/single.rs#L604

Reply:
Why are here so many unsafes